### PR TITLE
Allow filtering users by `created_at` or `updated_at` timestamp.

### PR DESCRIPTION
**Pull request by DFurnes**
#### What's this PR do?
This pull request adds support for querying users by timeframes. For example:

```sh
# users updated before 1/1/2010
GET v1/users?before[updated_at]=1/1/2010

# users updated after 1/1/2010
GET v1/users?after[updated_at]=1/1/2010

# users created between 1/1/2010 and 10/25/2011
GET v1/users?after[created_at]=1/1/2010&before[created_at]=10/25/2011
```

It also includes various little touch-ups, like improved docblocks & test helpers.

#### How should this be reviewed?
I'd recommend going commit-by-commit.

#### Checklist
- [ ] Documentation added for changed endpoints.
- [ ] Tests added for new features/bug fixes.
- [ ] Post a message in #api if this includes something that causes a rebuild!

### CODE REVIEW

**Comment by DFurnes** [comment]
[(5) As a data analyst, I want to be able to query Northstar for users by timeframe.](https://trello.com/c/w6aMPWG1/710-5-as-a-data-analyst-i-want-to-be-able-to-query-northstar-for-users-by-timeframe)

**Comment by sheyd** [comment]
@DFurnes Are date ranges queryable by hours, e.g. `GET v1/users?after[updated_at]=1/1/2010T12:00`?

**Comment by DFurnes** [comment]
@sheyd Yes! There's a fair number of ranges [supported](http://php.net/manual/en/datetime.formats.php), including ISO-8601 datetimes.

File: app/Auth/Normalizer.php [hunk 1]
```diff
@@ -72,6 +74,21 @@ public function mobile($mobile)
     }
 
     /**
+     * Parse a string into a Carbon date.
+     *
+     * @param string[] $strings
```
Comment (weerd) [inline app/Auth/Normalizer.php]: stray `[]`?

File: app/Auth/Normalizer.php [hunk 2]
```diff
@@ -72,6 +74,21 @@ public function mobile($mobile)
...
+     * @return string
```
Comment (weerd) [inline app/Auth/Normalizer.php]: Also, isn't the return an array and not a string?

File: app/Auth/Normalizer.php (hunk 1 above)
Comment (DFurnes) [inline app/Auth/Normalizer.php]: An [array of strings](https://phpdoc.org/docs/latest/guides/types.html#arrays)!

File: app/Auth/Normalizer.php (hunk 2 above)
Comment (DFurnes) [inline app/Auth/Normalizer.php]: Oops, fixed!

File: app/Auth/Normalizer.php (hunk 1 above)
Comment (weerd) [inline app/Auth/Normalizer.php]: Ah cool!

**Pull request merged**